pkg/datapath/linux/node_ids.go

```diff
@@ -144,16 +146,16 @@ func (n *linuxNodeHandler) deleteNodeIPs(oldIPs, newIPs []nodeTypes.Address) {
 	}
 }
 
-func (n *linuxNodeHandler) AllocateIDForNode(node *nodeTypes.Node) uint16 {
-	nodeID := n.allocateIDForNode(node)
+func (n *linuxNodeHandler) allocateIDForNode(node *nodeTypes.Node) uint16 {
+	nodeID := n.allocateNodeID(node)
 	return nodeID.id
 }
 
-// allocateIDForNode allocates a new ID for the given node if one hasn't already
+// allocateNodeID allocates a new ID for the given node if one hasn't already
 // been assigned. If any of the node IPs have an ID associated, then all other
 // node IPs receive the same. This might happen if we allocated a node ID from
 // the ipcache, where we don't have all node IPs but only one.
-func (n *linuxNodeHandler) allocateIDForNode(node *nodeTypes.Node) *nodeID {
+func (n *linuxNodeHandler) allocateNodeID(node *nodeTypes.Node) *nodeID {
```

Comment:
The idea was to have a function to allocate that returns the uint16 and a function to call to get the nodeID struct. Then we should only export the uint16 so that nodeid encapsulates the structure. Anyways I didn't do it very neatly so will clean it up and hopefully it becomes more obvious.